run-tests/action.yaml

```diff
@@ -0,0 +1,78 @@
+name: Run tests
+description: Install a snap and run specified tests for it.
+author: Snapcrafters
+branding:
+  icon: check
+  color: orange
+
+inputs:
+  channel:
+    description: "The channel to install the snap from"
+    default: "latest/candidate"
+    required: false
+  github-token:
+    description: "A token with permissions to comment on issues"
+    required: true
+  snapcraft-project-root:
+    description: "The root of the snapcraft project, where the `snapcraft` command would usually be executed from."
+    required: false
+  issue-number:
+    description: "Number of the issue comment on."
+    required: true
+  test-script:
+    description: "The script to run for testing."
+    required: true
+
+runs:
+  using: composite
+  steps:
+    - name: Checkout the code
+      uses: actions/checkout@v4
+
+    - name: Fetch build manifests
+      uses: snapcrafters/ci/fetch-manifests@main
+      with:
+        token: ${{ inputs.github-token }}
+
+    - name: Find and parse snapcraft.yaml
+      id: snapcraft-yaml
+      uses: snapcrafters/ci/parse-snapcraft-yaml@main
+      with:
+        snapcraft-project-root: ${{ inputs.snapcraft-project-root }}
+
+    - name: Install snap
+      shell: bash
+      env:
+        snap_name: ${{ steps.snapcraft-yaml.outputs.snap-name }}
+      run: |
+        # If we got a manifest file then parse the revision from it
+        if ls manifest-amd64.yaml &>/dev/null; then
+          rev="$(yq -r '.revision' manifest-amd64.yaml)"
+          echo "Installing snap revision '${rev}' from build manifest"
+          sudo snap install "${snap_name}" --revision "${rev}"
+        else
+          echo "Installing snap from '${{ inputs.channel }}'"
+          sudo snap install "${snap_name}" --channel "${{ inputs.channel }}"
+        fi
+
+    - name: Run tests
+      id: tests
+      shell: bash
+      run: |
+        # shellcheck disable=SC2288,SC2086,SC1083
+        ${{ inputs.test-script }}
```

Comment:
```suggestion
        ${{ inputs.test-script }} | tee test-output.txt
        echo "output=$(cat test-output.txt)" >> "$GITHUB_OUTPUT"
```

Reply:
I couldn't get multiline outputs to work nicely, so I used Github's step summary feature instead.